spi: Remove deprecated fields from struct spi_cs_control

galak opened the pull request:
All in tree users are using the gpio_dt_spec so we can remove the
older gpio struct elements that have been deprecated for some
time.

Signed-off-by: Kumar Gala <galak@kernel.org>

galak committed:
spi: Remove deprecated fields from struct spi_cs_control

All in tree users are using the gpio_dt_spec so we can remove the
older gpio struct elements that have been deprecated for some
time.

Signed-off-by: Kumar Gala <galak@kernel.org>

carlescufi commented:
> @carlescufi was curious about feeling on making this just:
> 
> ```
> struct spi_cs_control {
>    struct gpio_dt_spec gpio;
>    uint32_t delay;
> };
> ```
> 
> and if we consider that an API change?

I will bring it tomorrow in the Architecture WG

carlescufi commented:
Arch WG:

- @MaureenHelm will check whether the macro might still fail
- Barring any findings from @MaureenHelm remove the struct and mention in relnotes

galak committed:
spi: Remove anonymous struct from struct spi_cs_control

Simplify spi_cs_control by removing anonymous struct around the
"gpio" field.  Update SPI_CS_CONTROL_PTR_DT to match this change.

Signed-off-by: Kumar Gala <galak@kernel.org>

galak commented:
@MaureenHelm take a look and see if this builds ok w/XCC.  I don't think it should have any issues as we are removing the anonymous struct & union that it would have had issues with.

MaureenHelm reviewed (dismissed):
> @MaureenHelm take a look and see if this builds ok w/XCC. I don't think it should have any issues as we are removing the anonymous struct & union that it would have had issues with.

Confirmed that this builds ok with XCC.
```
$ west build -b intel_adsp_ace15_mtpm tests/drivers/spi/dt_spec
```

gmarull commented on doc/releases/release-notes-3.2.rst:
```diff
@@ -38,6 +43,9 @@ Removed APIs in this release
   * ``k_delayed_work_remaining_ticks()``
   * ``K_DELAYED_WORK_DEFINE``
 
+* Removed deprecated SPI ``struct spi_cs_control`` fields for GPIO management
+  that have been replaced with ``struct gpio_dt_spec``.
```
```suggestion
* Removed deprecated SPI :c:struct:`spi_cs_control` fields for GPIO management
  that have been replaced with :c:struct:`gpio_dt_spec`.
```

gmarull commented on doc/releases/release-notes-3.2.rst:
```diff
@@ -19,6 +19,11 @@ API Changes
 Changes in this release
 =======================
 
+* Changed ``struct spi_cs_control`` to remove anonymous struct.
```
```suggestion
* Changed :c:struct:`spi_cs_control` to remove anonymous struct.
```
